os/root-on-raid.md

```diff
@@ -0,0 +1,58 @@
+# Setting up Root on Raid
+
+Partitions used in RAID arrays need to have the gpt partition type uuid `be9067b9-ea49-4f15-b4f6-f36f8c9e1818` in order to boot. Because Ignition does not current support modifying existing partitions
+without wiping the partition table, the default ROOT partition cannot be used since it has different partition type uuid. In order to have the root partition in a raid array:
+ * Create the partitions used in the RAID array with the gpt partition type uuid `be9067b9-ea49-4f15-b4f6-f36f8c9e1818`
+ * Create a RAID array with the created partitions 
+ * Create a filesystem with on the raid array with the ROOT label
+ * Remove the ROOT label from the old root filesystem by wiping the existing filesystem or overwriting it
+
+## Example CLC
+
+```yaml container-linux-config
+storage:
+  disks:
+    - device: /dev/vdb
+      wipe_table: true
+      partitions:
+       - label: root1
+         number: 1
+         size: 256MiB
+         start: 0
+         type_guid: be9067b9-ea49-4f15-b4f6-f36f8c9e1818
+       - label: root2
+         number: 2
+         size: 256MiB
+         start: 0
+         type_guid: be9067b9-ea49-4f15-b4f6-f36f8c9e1818
+  raid:
+    - name: "my_raid_array"
+      level: "raid1"
+      devices:
+        - "/dev/vdb1"
+        - "/dev/vdb2"
+  filesystems:
+    - name: "ROOT"
+      mount:
+        device: "/dev/md/my_raid_array"
+        format: "ext4"
+        create:
+          options:
+            - "-L"
+            - "ROOT"
+    - name: "unused"
+      mount:
+        device: "/dev/vda9"
+        format: "ext4"
+        create:
+          options:
+            - "-L"
+            - "unused"
+          force: true
+```
+
+## Under the Hood
+Since Container Linux has an immutable initramfs, the initramfs must have a predetermined way for finding the root filesystem. If the root partition is in a RAID array, the raid array must be started in order to find the root filesystem. To avoid assembling unnecessary arrays (which may be assembled with other options from mdadm.conf), the partitions that are part of the root raid array must have the partition type uuid `be9067b9-ea49-4f15-b4f6-f36f8c9e1818`. A udev rule in the initramfs autoassembles all partitions with that type uuid.
```

Comment:
This reads like an explanation of a design decision.  That's not appropriate for user docs.  If someone was told that they need to configure a RAID, they shouldn't need to know what an immutable initramfs or an unnecessary array or udev is.

The user needs to know:

- What to expect for root RAID and how to set it up (it needs a special GUID, and mdadm.conf is not respected)
- What to expect for data RAID and how to set it up (automatically assembled during boot, subject to settings in mdadm.conf, and *must not* use the special GUID)

Reply:
Should we suggest they use the generic linux raid partition GUID or just not the special one? The [Ignition docs](https://coreos.com/ignition/docs/latest/examples.html) don't specify one

Reply:
1. We should fix the Ignition docs.
2. In the long run (hopefully not too long) the Ignition user docs should really be migrated to CLC docs, coreos/bugs#2089.
3. I don't think step-by-step instructions for data RAID are in scope, but we should have a quick summary of the differences and a link to the docs that currently exist.

Reply:
So we should sugguest they use the generic linux raid one (otherwise it will default to linux filesystem). I agree that we shouldn't give step by step instructions for data RAID, but if we are explicitly saying "Don't use the special guid" we should probably tell them a sane value to use instead (in addition to linking to relevent docs).

Reply:
Yes, the generic one.  Let's go ahead and provide the alternative GUID and see how it reads.